### PR TITLE
Update sampling_metadata.py

### PR DESCRIPTION
**Pull request by Aviadr-neureality**
Purpose
This PR fixes a crash in the non-GPU sampler caused by a change in the logits tensor dtype. Specifically, when logits are in float16, the temperature and top_p tensors—if not explicitly set to float32—can lead to computation errors or incompatibilities. This PR addresses [#21936](https://github.com/vllm-project/vllm/issues/21936) by hardcoding float32 dtype for temperature and top_p in the SamplingTensors structure to ensure consistent behavior across both GPU and CPU execution paths.

Test Plan
Run vllm using models that output float16 logits.

Set temperature and top_p to arbitrary values (e.g., temperature=0.7, top_p=0.9).

Verify that these values are correctly cast to float32 and propagated through the sampling logic.

Confirm no crashes occur on CPU (non-GPU) inference mode.

### CODE REVIEW

File: vllm/model_executor/sampling_metadata.py [hunk 1]
```diff
@@ -539,19 +539,19 @@ def from_lists(
...
+            dtype=torch.float32,
             pin_memory=pin_memory,
         )
         min_ps_t = torch.tensor(
             min_ps,
             device="cpu",
-            dtype=dtype,
+            dtype=torch.float32,
             pin_memory=pin_memory,
         )
```
[the remark this answers is not in the document]
Comment (Aviadr-neureality) [reply vllm/model_executor/sampling_metadata.py:556]: this will make the dtype parameter of the function irrelevant.
also what is the meaning of those params?

**Review by WoosukKwon — APPROVED** [review]

**Comment by hmellor** [comment]
I'm pretty sure this is causing `pytest tests/entrypoints/openai/test_transcription_validation.py -vsx -k sampling_params` to fail on main

**Comment by hmellor** [comment]
Yep, reverting this commit fixes that test

**Comment by Aviadr-neureality** [comment]
@hmellor , sorry guys, will fix the test ASAP.

**Comment by hmellor** [comment]
This is also causing the lm_eval tests such as `pytest tests/entrypoints/openai/correctness/test_lmeval.py -vsx -k test_lm_eval_accuracy_v0_engine[more_args0]` to fail too

**Comment by Aviadr-neureality** [comment]
@hmellor revert this, I’ll fix the tests and open a new PR.

**Comment by hmellor** [comment]
https://github.com/vllm-project/vllm/pull/22088

**Comment by WoosukKwon** [comment]
@hmellor Sorry, I didn't realized that the PR didn't run the full CI.

**Comment by Aviadr-neureality** [comment]
@WoosukKwon me too. Can i trigger it? Or it must run locally?
